dev/best-practice/using-tidb-in-java.md

```diff
@@ -0,0 +1,298 @@
+# Java 使用 TiDB 的最佳实践
```

Comment:
Please put this file under the `dev/reference/best-practices` directory, thanks~ 

I will review this PR later. :)